### DISCUSSION TITLE
How to open new API endpoints

### DISCUSSION THREAD

**Discussion by mrcyme** [BuchholzTim/Whitebird#91] [Q&A]
Hi, 
First of all congratulation for the amazing piece of work you put together.
For an application I am building, I have been looking for a whiteboard application that enables writing objects to the board with API calls.  As I could not find any board application providing such functionality (some does but not to the extent I am looking for), I looked for open source whiteboard applications and I came across yours. 
I was wondering if you have any piece of advice on what I should do to be able to write object to the board with an API calls. I just need to be able to add object in the board with an API call while keeping all other functionalities available. 

Thank you very much for the work you achieved !
Siméon

**Comment by BuchholzTim** [comment]
It's been a long time since I worked on the Application, so I hope it's correct what I'm saying:

The Application works mostly by using Socket-Events. All connected clients are connected to the backend by sockets. Whenever someone does something on the board a corresponding event is sent to the backend, which in turn sends another corresponding event to all other clients.

These events can be seen [here](https://github.com/BuchholzTim/Whitebird/blob/main/frontend/store/canvas.js) and [here](https://github.com/BuchholzTim/Whitebird/blob/main/backend/src/modules/socket/service/socket.gateway.ts)

You could possibly connect to the socket and send the appropriate events with the data you need and it should work.

**Comment by MarcBollmann** [comment]
Another alternative (the restful way) would be you write a new route in the nest-backend which could use the socket-gateway as a service. In your request you would send the fabricjs object and hand it over to the corresponding  socket-gateway method.
